applications/callflow/src/module/cf_log_in_out_ring_group.erl

```diff
@@ -0,0 +1,91 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% Logs a user in or out of a ring group
+%%% @end
+%%% @contributors
+%%%   Max Lay
+%%%-------------------------------------------------------------------
+-module(cf_log_in_out_ring_group).
+
+-behaviour(gen_cf_action).
+
+-include("callflow.hrl").
+
+-export([handle/2]).
+
+-define(ON_VAL, 0).
+-define(DEFAULT_OFF_VAL, 66269664000).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Entry point for this module, attempts to call an endpoint as defined
+%% in the Data payload.  Returns continue if fails to connect or
+%% stop when successfull.
+%% @end
+%%--------------------------------------------------------------------
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    _ = kapps_call_command:answer(Call),
+    CallflowId = kz_json:get_value(<<"callflow_id">>, Data),
+    DisableUntil = case kz_json:is_true(<<"login">>, Data) of
+                       'true' -> ?ON_VAL;
+                       'false' -> ?DEFAULT_OFF_VAL
+                   end,
+    AccountDb = kapps_call:account_db(Call),
+    {'ok', Callflow} = kz_datamgr:open_doc(AccountDb, CallflowId),
+    UpdatedFlow = find_and_change_ring_group(kz_json:get_value(<<"flow">>, Callflow), Call, DisableUntil),
+    kz_datamgr:save_doc(AccountDb, kz_json:set_value(<<"flow">>, UpdatedFlow, Callflow)),
```

Comment:
I didn't want to run the risk of overwriting changes to the callflow made elsewhere, but didn't know how to handle the potential failure to save the document. Any suggestions?